packages/devtools_app/lib/src/memory/memory_screen.dart

```diff
@@ -1076,64 +1170,28 @@ class MemoryBodyState extends State<MemoryBody>
     return results;
   }
 
+  List<Widget> displayVmDataInHover(ChartsValues chartsValues) =>
+      _dataToDisplay(
+        chartsValues.displayVmDataToDisplay(vmChartController.traces),
+      );
+
   List<Widget> displayAndroidDataInHover(ChartsValues chartsValues) {
-    const totalDisplay = 'Total';
-    const otherDisplay = 'Other';
-    const codeDisplay = 'Code';
-    const nativeDisplay = 'Native';
-    const javaDisplay = 'Java';
-    const stackDisplay = 'Stack';
-    const graphicsDisplay = 'Graphics';
+    if (!controller.isAndroidChartVisible) return [];
 
-    final results = <Widget>[];
+    final androidDataDisplayed =
+        chartsValues.androidDataToDisplay(androidChartController.traces);
 
-    if (controller.isAndroidChartVisible) {
-      final androidDataDisplayed = <String, String>{};
-
-      final data = chartsValues.androidData;
-
-      final totalValueDisplay = formatNumeric(data[adbTotalJsonName]);
-      androidDataDisplayed['$totalDisplay $totalValueDisplay'] =
-          androidTotalLegend;
-
-      final otherValueDisplay = formatNumeric(data[adbOtherJsonName]);
-      androidDataDisplayed['$otherDisplay $otherValueDisplay'] =
-          androidOtherLegend;
-
-      final codeValueDisplay = formatNumeric(data[adbCodeJsonName]);
-      androidDataDisplayed['$codeDisplay $codeValueDisplay'] =
-          androidCodeLegend;
-
-      final nativeValueDisplay = formatNumeric(data[adbNativeHeapJsonName]);
-      androidDataDisplayed['$nativeDisplay $nativeValueDisplay'] =
-          androidNativeLegend;
-
-      final javaValueDisplay = formatNumeric(data[adbJavaHeapJsonName]);
-      androidDataDisplayed['$javaDisplay $javaValueDisplay'] =
-          androidJavaLegend;
-
-      final stackValueDisplay = formatNumeric(data[adbStackJsonName]);
-      androidDataDisplayed['$stackDisplay $stackValueDisplay'] =
-          androidStackLegend;
-
-      final graphicsValueDisplay = formatNumeric(data[adbGraphicsJsonName]);
-      androidDataDisplayed['$graphicsDisplay $graphicsValueDisplay'] =
-          androidGraphicsLegend;
-
-      for (var entry in androidDataDisplayed.entries) {
-        results.add(
-          hoverRow(
-            name: entry.key,
-            image: entry.value,
-            hasNumeric: true,
-            hasUnit: controller.unitDisplayed.value,
-            scaleImage: true,
-          ),
-        );
-      }
-    }
+    // Separator between Android data.
+    const width = MemoryBodyState.hoverWidth - 40 - DashedLine.defaultDashWidth;
+    final dashedColor = Colors.grey.shade600;
 
-    return results;
+    return _dataToDisplay(
+      androidDataDisplayed,
+      firstWidget: Padding(
+        child: CustomPaint(painter: DashedLine(width, dashedColor)),
+        padding: const EdgeInsets.fromLTRB(20, 2, 20, 2),
```

Comment:
another use of interesting values. Can we use the default values that are used for padding elsewhere in DevTools? also, use `EdgeInsets.symmetric` if horizontal and vertical paddings are symmetrical

Reply:
```
const dividerLineVerticalSpace = 2.0;
const dividerLineHorizontalSpace = 20.0;

padding: const EdgeInsets.symmetric(
  vertical: dividerLineVerticalSpace,
  horizontal: dividerLineHorizontalSpace,
),
```